airbyte-ci/connectors/pipelines/tests/test_poetry/test_poetry_publish.py

```diff
@@ -0,0 +1,92 @@
+#
+# Copyright (c) 2023 Airbyte, Inc., all rights reserved.
+#
+
+import json
+import random
+from pathlib import Path
+from typing import List
+from unittest.mock import AsyncMock, MagicMock, patch
+
+import anyio
+import pytest
+import requests
+from connector_ops.utils import Connector, ConnectorLanguage
+from dagger import Client, Directory, Platform
+from pipelines.airbyte_ci.connectors.context import ConnectorContext
+from pipelines.airbyte_ci.connectors.publish import pipeline as publish_pipeline
+from pipelines.airbyte_ci.connectors.upgrade_cdk import pipeline as upgrade_cdk_pipeline
+from pipelines.airbyte_ci.poetry.publish.context import PyPIPublishContext
+from pipelines.dagger.actions.python.poetry import with_poetry
+from pipelines.models.contexts.pipeline_context import PipelineContext
+from pipelines.models.steps import StepStatus
+
+pytestmark = [
+    pytest.mark.anyio,
+]
+
+
+@pytest.fixture
+def context(dagger_client: Client):
+    context = PyPIPublishContext(
+        package_path="test",
+        version="0.2.0",
+        pypi_token="test",
+        package_name="test",
+        registry="http://local_registry:8080/",
+        is_local=True,
+        git_branch="test",
+        git_revision="test",
+        report_output_prefix="test",
+        ci_report_bucket="test",
+    )
+    context.dagger_client = dagger_client
+    return context
+
+
+@pytest.mark.parametrize(
+    "package_path, package_name, expected_asset",
+    [
+        pytest.param(
+            "airbyte-integrations/connectors/source-apify-dataset",
+            "airbyte-source-apify-dataset",
+            "airbyte_source_apify_dataset-0.2.0-py3-none-any.whl",
+            id="setup.py project",
+        ),
+        pytest.param(
+            "airbyte-integrations/connectors/destination-duckdb",
+            "destination-duckdb",
+            "destination_duckdb-0.2.0-py3-none-any.whl",
+            id="poetry project",
+        ),
+    ],
+)
+async def test_run_poetry_publish(context: PyPIPublishContext, package_path: str, package_name: str, expected_asset: str):
```

Comment:
Would it be worth adding the same kind of integration test for "non poetry" package?

Reply:
apify-dataset is a `setup.py`-style package and airbyte-lib is a poetry-style package, so both branches are tested via the parameterization.